pkgs/by-name/lc/lc0/package.nix

```
@@ -0,0 +1,69 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, meson
+, ninja
+, pkg-config
+, python3
+, cmake
+, zlib
+, gtest
+, eigen
+, ...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "lc0";
+  version = "0.30.0";
+
+  src = fetchFromGitHub {
+    owner = "LeelaChessZero";
+    repo = "lc0";
+    rev = "refs/tags/v${version}";
+    sha256 = "sha256-Q85hXNSexHhk6W48tgJLk0Sf32xBipfg2P2SH1FF89Q=";
+    fetchSubmodules = true;
+  };
+
+  patchPhase = ''
+    patchShebangs --build /build/source/scripts/*
+  '';
+
+  nativeBuildInputs = [
+    meson
+    cmake
+    ninja
+    pkg-config
+    python3
+    zlib
+    gtest
+  ];
+
+  mesonFlags = [
+    "-Dplain_cuda=false"
+    "-Daccelerate=false"
+    "-Dmetal=disabled"
+    "-Dembed=false"
+  ] ++
+  # in version 31 this option will be required
+  (if version == "0.30.0" then [ ] else [ "-Dnative_cuda=false" ]);
+
+
```

```suggestion
  ]
  # in version 31 this option will be required
  ++ lib.optionals (lib.versionAtLeast "0.31") [ "-Dnative_cuda=false" ];
```

Alternatively, we could just add this in later:
```suggestion
  ];
```